gobblin-runtime/src/main/java/gobblin/runtime/MultiConverter.java

```diff
@@ -80,87 +79,122 @@ public Object convertSchema(Object inputSchema, WorkUnitState workUnit)
   }
 
   /**
-   * A type of {@link java.util.Iterator} to be used with {@link MultiConverter}.
+   * A type of {@link java.util.Iterator} to be used with {@link MultiConverter}. The Converter uses the
+   * {@link ChainedConverterIterator} to chain iterators together. The first ChainedConverterIterator created contains
+   * an iterator with only the inputRecord and the first converter in the converters list. Each subsequent
+   * ChainedConverterIterator is created using the previous ChainedConverterIterator along with the next converter in
+   * the converters list. By chaining the converters and iterators in this fashion, a reference to the last
+   * ChainedConverterIterator will be sufficient to iterate through all the data.
    */
   private class MultiConverterIterator implements Iterator<Object> {
 
     private final WorkUnitState workUnitState;
-    private final Deque<Iterator<Object>> converterIteratorStack = Lists.newLinkedList();
-    private Iterator<Object> outputIterator;
+
+    private Iterator<Object> chainedConverterIterator;
```

Comment:
I re-assign it in the constructor multiple times via the for loop, the compiler wouldn't let me make it final